CSS transition on SVG inside href fails when link already visited

anttijk opened the pull request:
#### 969e1262811073faa52aa31cbceea18ef64183d8
<pre>
CSS transition on SVG inside href fails when link already visited
<a href="https://bugs.webkit.org/show_bug.cgi?id=267515">https://bugs.webkit.org/show_bug.cgi?id=267515</a>
<a href="https://rdar.apple.com/121015467">rdar://121015467</a>

Reviewed by Antoine Quint.

* LayoutTests/transitions/svg-visited-fill-expected.html: Added.
* LayoutTests/transitions/svg-visited-fill.html: Added.
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::PropertyWrapperVisitedAffectedSVGPaint::PropertyWrapperVisitedAffectedSVGPaint):
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):

Add an animation property wrapper for visited-affected fill/stroke similar to other color properties and use it.

* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/SVGRenderStyle.h:
(WebCore::RenderStyle::fillPaintColor const):
(WebCore::RenderStyle::visitedFillPaintColor const):
(WebCore::RenderStyle::visitedFillPaintType const):
(WebCore::RenderStyle::setFillPaintColor):
(WebCore::RenderStyle::setVisitedFillPaintColor):
(WebCore::RenderStyle::setStrokePaintColor):
(WebCore::RenderStyle::setVisitedStrokePaintColor):
(WebCore::RenderStyle::strokePaintColor const):
(WebCore::RenderStyle::visitedStrokePaintColor const):
(WebCore::RenderStyle::visitedStrokePaintType const):

Expose some helpers for the property wrapper.

Canonical link: <a href="https://commits.webkit.org/279807@main">https://commits.webkit.org/279807@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cc4192a79223c1e65c62f296bd40eb7a588b9e34

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54608 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34027 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7185 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57889 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5342 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56910 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41567 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5345 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44230 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3597 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56703 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32172 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47302 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25356 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28969 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4633 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3482 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50719 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4849 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59479 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29843 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4985 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51652 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30991 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47388 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51029 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11947 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31970 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30779 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

graouts reviewed (approved):
Your commit message only contains the test changes, make sure to update that prior to landing.

anttijk committed:
CSS transition on SVG inside href fails when link already visited
https://bugs.webkit.org/show_bug.cgi?id=267515
rdar://121015467

Reviewed by Antoine Quint.

* LayoutTests/transitions/svg-visited-fill-expected.html: Added.
* LayoutTests/transitions/svg-visited-fill.html: Added.
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::PropertyWrapperVisitedAffectedSVGPaint::PropertyWrapperVisitedAffectedSVGPaint):
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):

Add an animation property wrapper for visited-affected fill/stroke similar to other color properties and use it.

* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/SVGRenderStyle.h:
(WebCore::RenderStyle::fillPaintColor const):
(WebCore::RenderStyle::visitedFillPaintColor const):
(WebCore::RenderStyle::visitedFillPaintType const):
(WebCore::RenderStyle::setFillPaintColor):
(WebCore::RenderStyle::setVisitedFillPaintColor):
(WebCore::RenderStyle::setStrokePaintColor):
(WebCore::RenderStyle::setVisitedStrokePaintColor):
(WebCore::RenderStyle::strokePaintColor const):
(WebCore::RenderStyle::visitedStrokePaintColor const):
(WebCore::RenderStyle::visitedStrokePaintType const):

Expose some helpers for the property wrapper.

Canonical link: https://commits.webkit.org/279807@main
